Resources/template.json

```diff
@@ -282,7 +299,43 @@
       ],
       "copy": {
         "name": "keyvaultReadAccessCopy",
-        "count": "[length(parameters('defaultKeyVaultReadAccess'))]",
+        "count": "[variables('defaultKeyVaultReadAccessHack')]",

```

Comment:
Isn't this variable a bool rather than an int?

Reply:
No, it's a number:

```
    "defaultKeyVaultReadAccessHack": "[if(empty(parameters('defaultKeyVaultReadAccess')), 1, length(parameters('defaultKeyVaultReadAccess')))]",
```

If the array is empty, it places 1 in the variable,  otherwise it places the true length of the parameter in there.

There's a condition on the deployment where it will only trigger if is more than 1 entry in the 'defaultKeyVaultReadAcccess' parameter.

But this fails when it's empty, because ARM errors if you have a copyIndex count of 0.

So, we use the 'defaultKeyVaultReadAccessHack' variable as a work-around for the copyIndex.  When there are no entries in the parameter,  it places a value of 1 into the copyIndex, and the block is never executed due to the condition.
When there are entries in the parameter, it executes as you'd expect.